features/testbot/traffic_speeds.feature

```diff
@@ -48,10 +48,10 @@ Feature: Traffic - speeds
 
         When I route I should get
           | from | to | route       | speed   | weights              | a:datasources |
-          | a    | b  | ad,de,eb,eb | 30 km/h | 1275.7,400.4,378.2,0 | 1:0:0:0       |
+          | a    | b  | ad,de,eb,eb | 30 km/h | 1275.7,400.4,378.2,0 | 1:0:0         |
```

Comment:
This seems wrong: `a:datasources` should have the same length as `weights`.

Reply:
@TheMarex the test case already has an inconsistency in expectation values: lines 53-54 have two weight values and two datasources, lines 55-56 have two weight values and one datasource.  The change only fixes  inconsistency in annotation values: now number of annotation weights, durations, data sources, speed values is by one less than number of node IDs. I'll :eyes: how to make consistent step `weights` and annotation `a:weight`, because with `master` number of weights is different for the `dc` case:
```
| from | to | route       | speed   | weights              | a:weight             |
| a    | b  | ad,de,eb,eb | 30 km/h | 1275.7,400.4,378.2,0 | 1275.7:400.4:378.2:0 |
| b    | c  | bc,bc       | 27 km/h | 741.5,0              | 741.5:0              |
| d    | c  | dc,dc       | 36 km/h | 956.8,0              | 956.8                |
```

Reply:
Oh I see so actually the `weights` are wrong in this scenario?

Reply:
I'm 👍 on just fixing one problem at a time here.